src/tree.rs

```diff
@@ -1,66 +1,91 @@
 //! The baseline requirements of any UI Tree so Taffy can efficiently calculate the layout
 
-use slotmap::DefaultKey;
-
 use crate::{
-    error::TaffyResult,
-    layout::{Cache, Layout},
+    layout::{Cache, Layout, SizeAndBaselines, SizingMode},
     prelude::*,
 };
+use core::fmt::Debug;
 
 /// Any item that implements the LayoutTree can be layed out using Taffy's algorithms.
 ///
 /// Generally, Taffy expects your Node tree to be indexable by stable indices. A "stable" index means that the Node's ID
 /// remains the same between re-layouts.
 pub trait LayoutTree {
```

Comment:
> We should have some examples showing how (and when) to use the various methods on the trait, especially if it will be up to the end user to run the layout on each of the node

Agreed. Alos, I've just noticed that this comment (about stable indexes) is no longer accurate.